client/network/src/protocol/sync.rs

```diff
@@ -606,10 +650,13 @@ impl<B: BlockT> ChainSync<B> {
 		let best_queued = self.best_queued_number;
 		let client = &self.client;
 		let queue = &self.queue_blocks;
+		let pending_requests = self.pending_requests.take();
```

Comment:
There's a small logic issue: at the moment the code in `protocol.rs` calls `block_request`, then grabs the first element of the returned iterator, and immediately returns `Poll::Ready`, which drops the rest.

(Then afterwards poll is called again, and we call `block_requests` again to get a new iterator).

If this gets in, then we will also forever discard all the requests except the first.

In #5824 I also changed that so that we immediately add all the requests to `pending_events`, and this could be applied here.
But on the other hand, it's also maybe not fool-proof that dropping the iterator early also drops all the elements that were not iterated over?
I don't have a strong opinion, and doing the same as #5824 would be good for me.